test/e2e/cruise/basic_test.go

```diff
@@ -254,18 +260,27 @@ var _ = Describe("Liqo E2E", func() {
 				}
 			)
 
+			BeforeEach(func() {
+				options = k8s.NewKubectlOptions("", testContext.Clusters[0].KubeconfigPath, namespace)
+			})
+
 			AfterEach(func() {
-				Expect(testContext.Clusters[0].NativeClient.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})).To(Succeed())
+				Expect(client.IgnoreNotFound(testContext.Clusters[0].NativeClient.CoreV1().Namespaces().
+					Delete(ctx, namespace, metav1.DeleteOptions{}))).To(Succeed())
+
+				Eventually(func() bool {
+					_, err := testContext.Clusters[0].NativeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+					return apierrors.IsNotFound(err)
+				}, timeout, interval).Should(BeTrue())
```

Comment:
When possible, it is better to return the un-processed variable instead of a boolean value, since then it is printed in full in case of error. Could then use the `BeNotFound()` helper.